src/Microsoft.Data.SqlClient/src/Microsoft/Data/Common/DbConnectionStringCommon.cs

```diff
@@ -361,7 +361,8 @@ internal static ApplicationIntent ConvertToApplicationIntent(string keyword, obj
             "ActiveDirectoryDeviceCodeFlow",
             "ActiveDirectoryManagedIdentity",
             "ActiveDirectoryMSI",
-            "ActiveDirectoryDefault"
+            "ActiveDirectoryDefault",
+            "ActiveDirectoryTokenCredential"
```

Comment:
We don't need to add a new Authentication option. We can simply key off whether AccessTokenCallback is null or not.
```suggestion
            "ActiveDirectoryDefault"
```